lua/nightfox/init.lua

```diff
@@ -1,71 +1,118 @@
 local M = {}
 
-local override = {}
-
-function override.palettes(palettes)
-  require("nightfox.override").palettes = palettes
+function M.compile()
+  local compiler = require("nightfox.lib.compiler")
+  local foxes = require("nightfox.palette").foxes
+  for _, style in ipairs(foxes) do
+    compiler.compile({ style = style })
+  end
 end
 
-function override.specs(specs)
-  require("nightfox.override").specs = specs
+function M.reset()
+  require("nightfox.config").reset()
+  require("nightfox.override").reset()
 end
 
-function override.groups(groups)
-  require("nightfox.override").groups = groups
-end
+-- Avold g:colors_name reloading
+local lock = false
 
-M.override = override
+function M.load(opts)
+  if lock then
+    return
+  end
 
-function M.init(opts)
-  require("nightfox.config").set_options(opts)
-end
+  opts = opts or {}
 
-function M.compile()
-  require("nightfox.lib.compile").compile()
-end
+  local config = require("nightfox.config")
+  local _, compiled_file = config.get_compiled_info(opts)
+  lock = true
 
-function M.clean()
-  require("nightfox.lib.compile").clean()
-end
+  local ok, f = pcall(loadfile, compiled_file)
```

Comment:
pcall is slower. Just check if f is null

Reply:
I think I had some failures when working on the feature that made me wrap the load and call in a pcall. Now that the feature is complete you are right I dont need the pcall.